lib/host.rb

```diff
@@ -511,31 +511,24 @@ def delete(file, opts={})
         if bad_files.include? file
           raise "should not remove file named '#{file}'"
         end
-
-        r = "-r"
+        meth = :rm
       else
-        r = ""
+        meth = :remove
       end
       file = shell_escape file
       if opts[:home] && ! file.start_with?('/','\\')
         # relative to host local user home directory
         # relies strongly on bad_files checking above
         file = '~/' + file
       end
-      if opts[:raw]
-        exec_method = :exec_raw
-      elsif opts[:admin]
-        exec_method = :exec_admin
+      if File.file? file
+        FileUtils.public_send(method, file, force: true)
+        check_cmd = File.file? file
       else
-        exec_method = :exec
+        FileUtils.public_send(:remove_dir, file, force: true)
+        check_cmd = File.directory? file
       end
-      public_send(exec_method, "rm #{r} -f -- #{file}", **opts)
-      opts[:quiet] = true
-      res = public_send(exec_method, "ls -d -- #{file}", **opts)
-
-      # OCDebugAccessibleHost does not return exit status of executed command
```

Comment:
we still need something in OCDebugAccessibleHost to handle cleaning up in the chroot.

```
self = {BushSlicer::OCDebugAccessibleHost} root@ip-10-0-187-176
bad_files = Array (8 elements)
exec_method = :exec_raw
file = "/tmp/workdir/rbrattai-rbrattai"
opts = Hash (2 elements)
 :r => true
 :raw => true
r = "-r"
res = nil
```

It looks like we can use Ruby methods for `LocalLinuxLikeHost` but we have to use shell for `SSHAccessibleHost` and  `OCDebugAccessibleHost`.

Maybe we can move methods around between the classes.

Reply:
OCDebugAccessibleHost test case

```
  @admin
  @destructive
  Scenario: net.ipv4.ip_forward should be always enabled on node service startup
    Given I select a random node's host

    When I run commands on the host:
      | sysctl -w net.ipv4.ip_forward=0 |
    Then the step should succeed
```

Reply:
@rbbratta does that testcase call the `delete` methods?  I just ran it and it passed.

Reply:
yes, the original call trace when using `When I run commands on the host:` is

```
delete [host.rb:532] (BushSlicer::LinuxLikeHost)
user_opts [cli_executor.rb:287] (BushSlicer::SharedLocalCliExecutor)
exec [cli_executor.rb:303] (BushSlicer::SharedLocalCliExecutor)
cli_exec [api_accessor.rb:71] (BushSlicer::APIAccessor)
get_matching [cluster_resource.rb:162] (BushSlicer::ClusterResource)
nodes [environment.rb:412] (BushSlicer::OpenShiftEnvironment)
node.rb:29

```

Reply:
it passes because `if File.file? file`  fails because the file is on the remote `oc debug` host so it calls `FileUtils.public_send(:remove_dir, file, force: true)` instead, and the `force: true` eats the error